client/my-sites/stats/activity-log/index.jsx

```diff
@@ -95,6 +95,34 @@ class ActivityLog extends Component {
 		window.scrollTo( 0, 0 );
 	}
 
+	getStartMoment() {
+		const { gmtOffset, moment, startDate, timezone } = this.props;
+
+		// Use current site time if no startDate is supplied
+		if ( ! startDate ) {
+			return this.applySiteOffset( moment(), true );
+		}
+
+		if ( timezone ) {
+			return moment.tz( startDate, timezone );
+		}
+		if ( null !== gmtOffset ) {
```

Comment:
Haven't really looked into how this can be done, but do you think we can simplify the `gmtOffset` handling code, probably by using `moment.utcOffset` or something?

Reply:
Trying to parse string dates with offsets is turning out to be really complicated.

There are known issues working with offsets and I've gotten a PR ready to work on that (#17320). It's quite complicated and I'd prefer avoid tackling that in this PR.